conans/client/cmake.py

```diff
@@ -115,18 +118,24 @@ def flags(self):
         op_system = str(self._settings.os) if self._settings.os else None
         arch = str(self._settings.arch) if self._settings.arch else None
         comp = str(self._settings.compiler) if self._settings.compiler else None
-        comp_version = self._settings.compiler.version
+        comp_version = self._settings.compiler.version     
+
+        try:
+            op_system_version = str(self._settings.os.version) if self._settings.os.version else None
```

Comment:
The os.version (aka `CMAKE_SYSTEM_VERSION`) is required when cross-compiling for specific operating systems (like for Windows and Windows CE, see the link to [CMake](https://cmake.org/cmake/help/v3.7/manual/cmake-toolchains.7.html#cross-compiling-for-windows-ce) docs). 

In fact, I guarded it with try/catch because not every operating system requires it.

About the tests, I will try to fix them by the deadline